RecoEgamma/EgammaPhotonProducers/src/ReducedEGProducer.cc

```diff
@@ -326,152 +343,126 @@ ReducedEGProducer::ReducedEGProducer(const edm::ParameterSet& config)
     setToken(gsfElectronCalibEcalEnergyErrT_, config, "gsfElectronCalibEcalEnergyErrSource");
   }
 
-  produces<reco::PhotonCollection>(outPhotons_);
-  produces<reco::PhotonCoreCollection>(outPhotonCores_);
   if (!ootPhotonT_.isUninitialized()) {
-    produces<reco::PhotonCollection>(outOOTPhotons_);
-    produces<reco::PhotonCoreCollection>(outOOTPhotonCores_);
-  }
-  produces<reco::GsfElectronCollection>(outGsfElectrons_);
-  produces<reco::GsfElectronCoreCollection>(outGsfElectronCores_);
-  produces<reco::GsfTrackCollection>(outGsfTracks_);
-  produces<reco::ConversionCollection>(outConversions_);
-  produces<reco::ConversionCollection>(outSingleConversions_);
-  produces<reco::SuperClusterCollection>(outSuperClusters_);
-  produces<reco::CaloClusterCollection>(outEBEEClusters_);
-  produces<reco::CaloClusterCollection>(outESClusters_);
-  if (!ootPhotonT_.isUninitialized()) {
-    produces<reco::SuperClusterCollection>(outOOTSuperClusters_);
-    produces<reco::CaloClusterCollection>(outOOTEBEEClusters_);
-    produces<reco::CaloClusterCollection>(outOOTESClusters_);
+    outOOTPhotons_ = produces<reco::PhotonCollection>("reducedOOTPhotons");
+    outOOTPhotonCores_ = produces<reco::PhotonCoreCollection>("reducedOOTPhotonCores");
   }
-  produces<EcalRecHitCollection>(outEBRecHits_);
-  produces<EcalRecHitCollection>(outEERecHits_);
-  if (doPreshowerEcalHits_)
-    produces<EcalRecHitCollection>(outESRecHits_);
-  produces<HBHERecHitCollection>(outHBHERecHits_);
-  produces<edm::ValueMap<std::vector<reco::PFCandidateRef>>>(outPhotonPfCandMap_);
-  produces<edm::ValueMap<std::vector<reco::PFCandidateRef>>>(outGsfElectronPfCandMap_);
-  for (const std::string& outid : outPhotonIds_) {
-    produces<edm::ValueMap<bool>>(outid);
-  }
-  for (const std::string& outid : outGsfElectronIds_) {
-    produces<edm::ValueMap<float>>(outid);
+  if (!ootPhotonT_.isUninitialized()) {
```

Comment:
it seems to me that this may be merged with the if/else just above?

Reply:
Done.